static/js/components/NonMemberGroupList.jsx

```diff
@@ -0,0 +1,139 @@
+import { showNotification } from "baselayer/components/Notifications";
+
+import * as groupAdmissionRequestsActions from "../ducks/groupAdmissionRequests";
+
+const NonMemberGroupList = ({ groups }) => {
+  const dispatch = useDispatch();
+  const { id: currentUserID, groupAdmissionRequests } = useSelector(
+    (state) => state.profile
+  );
+  if (currentUserID === null) {
+    return <>Loading...</>;
+  }
+  const pendingRequestGroupIDs = groupAdmissionRequests
+    .filter((request) => request.status === "pending")
+    .map((request) => request.group_id);
+  const declinedRequestGroupIDs = groupAdmissionRequests
+    .filter((request) => request.status === "declined")
+    .map((request) => request.group_id);
+
+  const handleRequestAdmission = async (groupID) => {
+    const result = await dispatch(
+      groupAdmissionRequestsActions.requestGroupAdmission(
+        currentUserID,
+        groupID
+      )
+    );
+    if (result.status === "success") {
+      dispatch(showNotification("Successfully requested admission to group."));
+      // dispatch(groupsActions.fetchGroups(true));
+    }
+  };
+
+  const handleDeleteAdmissionRequest = (admissionRequestID) => {
+    dispatch(
+      groupAdmissionRequestsActions.deleteAdmissionRequest(admissionRequestID)
+    );
+  };
+
+  const renderActions = (dataIndex) => {
+    const group = groups[dataIndex];
+    if (declinedRequestGroupIDs.includes(group.id)) {
+      return <em>Admission request declined.</em>;
+    }
+    if (pendingRequestGroupIDs.includes(group.id)) {
+      const admissionRequestID = groupAdmissionRequests.filter(
+        (request) => request.group_id === group.id
+      )[0].id;
+      return (
+        <>
+          <em>Request pending...</em>
+          <br />
+          <Button
+            size="small"
+            variant="contained"
+            onClick={() => handleDeleteAdmissionRequest(admissionRequestID)}
+            data-testid={`deleteAdmissionRequestButton${group.id}`}
+          >
+            Delete request
+          </Button>
+        </>
+      );
+    }
+    return (
+      <Button
+        size="small"
+        variant="contained"
+        onClick={() => handleRequestAdmission(group.id)}
+        data-testid={`requestAdmissionButton${group.id}`}
+      >
+        Request admission
+      </Button>
+    );
+  };
+  const columns = [
+    {
+      name: "name",
+      label: "Name",
+      options: {
+        filter: false,
+      },
+    },
+    {
+      name: "nickname",
+      label: "Nickname",
+      options: {
+        filter: false,
+      },
+    },
+    {
+      name: "actions",
+      label: "Actions",
+      options: {
+        sort: false,
+        customBodyRenderLite: renderActions,
+        filter: false,
+      },
+    },
+  ];
+
+  const options = {
+    responsive: "standard",
+    download: false,
+    search: true,
+    selectableRows: "none",
+    rowsPerPage: 10,
+    rowsPerPageOptions: [10, 25, 50, 100, 200],
+    filter: true,
```

Comment:
```suggestion
    filter: false,
```
Don't think we need to have filtering for any of these columns (which is already the case), so just turn it off here to get rid of the toolbar icon.

Reply:
:+1:

Reply:
Done